### DISCUSSION TITLE
Allow users to select number of peaks to keep instead of stringency thresholds.

### DISCUSSION THREAD

**Discussion by callum-b** [Genometric/MSPC#216] [Ideas]
One setup I have used in the past is to call a set number of peaks for different experiments (for example, binding of a TF in various conditions) and then comparing how those peaks overlap on genomic features. An option in MSPC to keep the 50000 top peaks as determined by the combined p-value would allow this quite easily.

I know this can be done by setting a very permissible threshold and then filtering the output files, but if there is a way to do this during normal MSPC runtime, it would avoid sorting a massive bed file to find the top peaks.

**Comment by VJalili** [comment]
Agree, a parameter such as `--limit` could be helpful, similar to directives on SQL queries.  I wonder what would be minimal and comprehensive design.

- One `--limit` argument that is applied to all the outputs MSPC generates. 
- One `--limit` per output type. 

The latter gives you finer level control, but it would convoluted CLI options. The former is limited, but covers most use-cases.
